extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iProcessor.java

```diff
@@ -102,11 +109,30 @@ public void s2iRequirementsNative(S2iConfig s2iConfig,
             CurateOutcomeBuildItem curateOutcomeBuildItem,
             OutputTargetBuildItem out,
             PackageConfig packageConfig,
+            NativeImageBuildItem nativeImage,
             BuildProducer<BaseImageInfoBuildItem> builderImageProducer,
             BuildProducer<KubernetesCommandBuildItem> commandProducer) {
 
+        boolean usingDefaultBuilder = ImageUtil.getRepository(S2iConfig.DEFAULT_BASE_NATIVE_IMAGE)
+                .equals(ImageUtil.getRepository(s2iConfig.baseNativeImage));
+        String outputNativeBinaryFileName = nativeImage.getPath().getFileName().toString();
+
+        String nativeBinaryFileName = null;
+
+        //The default s2i builder for native builds, renames the native binary.
```

Comment:
wait - so our own native s2i builder goes an rename to something even more different ?

Reply:
I also think this shouldn't happen. It was probably a shortcut to make the `run` script easier. If anything, it shouldn't be a rename but a symlink.

Reply:
I logged an issue: https://github.com/quarkusio/quarkus-images/issues/58
Until the issue is address we need this, so I'll mark this discussion as resolved.